lib/user-ids-autom8-able.rb

```diff
@@ -121,6 +121,12 @@ def hard_delete(user_id, url, log_file)
             log_file.puts message
             next
 
+          rescue ZendeskAPI::Error::ReadTimeout => api_error
```

Comment:
I take it this kind of "best effort" approach to deleting the users is acceptable because the job is just going to run again later/another-day anyway, and that's "good enough"?

My concern is that in an event where network connectivity is blocked, this job will just happily plough on and end up green/sucessfully even tho it didn't actually do anything. (This can also be said for the NetworkError case too).

Reply:
Thanks @chrisfarms - good point. A little info: the job runs every night so a failure one night should be resolved the next night.
However you are absolutely correct in that if we handle all errors and just continue and it fails continuously and ends up green, we would not be any the wiser.
I will have a think and maybe we can discuss offline - maybe some count the errors thing which exits out creating an alert at a predetermined number.

Reply:
A rule such as "if we had one of these failures two nights in a row, throw an error" would work, but I don't know how you'd achieve this with concourse - there doesn't seem to be a way of registering "success, but with warnings".